SQLite/Foundation.swift

```diff
@@ -47,7 +47,7 @@ extension NSDate : Value {
     }
 
     public class func fromDatatypeValue(stringValue: String) -> NSDate {
-        return dateFormatter.dateFromString(stringValue)!
+        return dateFormatter.dateFromString(stringValue) ?? NSDate()
```

Comment:
Hi, a cleaner way to do this could be:

    public class func fromDatatypeValue(stringValue: String) -> NSDate? {
        return dateFormatter.dateFromString(stringValue)
    }

On 19 Jul 2016, at 21:21, Nick Shelley <notifications@github.com<mailto:notifications@github.com>> wrote:


In SQLite/Foundation.swift<https://github.com/stephencelis/SQLite.swift/pull/436#discussion_r71415629>:

> @@ -47,7 +47,7 @@ extension NSDate : Value {
>      }
>
>      public class func fromDatatypeValue(stringValue: String) -> NSDate {
> -        return dateFormatter.dateFromString(stringValue)!
> +        return dateFormatter.dateFromString(stringValue) ?? NSDate()


The problem with this is that it makes it seem like the database is actually storing a string that represents the current date, which can be misleading and confusing to users.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/stephencelis/SQLite.swift/pull/436/files/76c6e443a199aef3bb19709c7320520305ca1fc0#r71415629>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ASoNb8jTQBBLMuqMkUsUVX5afC1vlSUSks5qXTHYgaJpZM4Iki5->.